### PR TITLE
Perf: add extra index to notification table

### LINKED ISSUE #32390: Notification page becomes very slow when the table is very large
### Description

I hosted a Gitea instance with 1410973 rows in the notification table. Now Gitea admin needs ~6000ms to open the notification page. The slow SQL log is as follows:

```
Oct 31 14:50:44 git-server gitea[2883158]: 2024/10/31 14:50:44 models/db/list.go:164:]() [W] [Slow SQL Query] SELECT `id`, `user_id`, `repo_id`, `status`, `source`, `issue_id`, `commit_id`, `comment_id`, `updated_by`, `created_unix`, `updated_unix` FROM `notification` WHERE notification.user_id=? AND notification.status IN (?,?) ORDER BY notification.updated_unix DESC LIMIT 20 [70 1 3] - 5.11404742s
```

I tried to add an index in MySQL to temporarily fix it, `CREATE INDEX idx_notification_user_status_updated ON notification(user_id, status, updated_unix DESC);`. The notification page now only needs 6ms to load.

Should I do something other than adding the index manually to the table or can I create a PR to add the index to the code base?

### Gitea Version

1.22.0

### Can you reproduce the bug on the Gitea demo site?

No

### Log Gist

_No response_

### Screenshots

_No response_

### Git Version

_No response_

### Operating System

_No response_

### How are you running Gitea?

Use my own build from https://github.com/BoYanZh/gitea.

### Database

MySQL/MariaDB

### PR DESCRIPTION
**Pull request by BoYanZh**
Index SQL: `CREATE INDEX u_s_uu ON notification(user_id, status, updated_unix);`

The naming follows `action.go` in the same dir.

I am unsure which version I should add SQL to the migration folder, so I have not modified it.

Fix #32390

### CODE REVIEW

File: models/activities/notification.go [hunk 1]
```diff
@@ -71,6 +72,15 @@ type Notification struct {
 	UpdatedUnix timeutil.TimeStamp `xorm:"updated INDEX NOT NULL"`
 }
 
+// TableIndices implements xorm's TableIndices interface
+func (n *Notification) TableIndices() []*schemas.Index {
```
Comment (lunny) [inline models/activities/notification.go:76]: Once you created the `TableIndices` method, please remove all tags on the struct for the xorm index.

File: models/migrations/v1_23/v307.go [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+	IssueID   int64  `xorm:"INDEX NOT NULL"`
+	CommitID  string `xorm:"INDEX"`
+	CommentID int64
+
+	UpdatedBy int64 `xorm:"INDEX NOT NULL"`
+
+	Issue      *issues_model.Issue    `xorm:"-"`
+	Repository *repo_model.Repository `xorm:"-"`
+	Comment    *issues_model.Comment  `xorm:"-"`
```
Comment (lunny) [inline models/migrations/v1_23/v307.go]: These lines should be removed.

File: models/migrations/v1_23/v307.go [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+	UpdatedUnix timeutil.TimeStamp `xorm:"updated INDEX NOT NULL"`
+}
+
+// TableName sets the name of this table
+func (*improveNotificationTableIndicesAction) TableName() string {
+	return "notification"
+}
+
+// TableIndices implements xorm's TableIndices interface
+func (*improveNotificationTableIndicesAction) TableIndices() []*schemas.Index {
```
Comment (lunny) [inline models/migrations/v1_23/v307.go]: We need the full indexes.

File: models/migrations/v1_23/v307.go [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+import (
+	"code.gitea.io/gitea/modules/timeutil"
+
+	"xorm.io/xorm"
+	"xorm.io/xorm/schemas"
+)
+
+type improveNotificationTableIndicesAction struct {
+	ID     int64 `xorm:"pk autoincr"`
+	UserID int64 `xorm:"INDEX NOT NULL"`
```
Comment (lunny) [inline models/migrations/v1_23/v307.go]: INDEX could be removed here.

File: models/migrations/v1_23/v307.go [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+	RepoID int64 `xorm:"INDEX NOT NULL"`
+
+	Status uint8 `xorm:"SMALLINT INDEX NOT NULL"`
+	Source uint8 `xorm:"SMALLINT INDEX NOT NULL"`
+
+	IssueID   int64  `xorm:"INDEX NOT NULL"`
```
Comment (lunny) [inline models/migrations/v1_23/v307.go]: The same as above.

File: models/migrations/v1_23/v307.go [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+	CommitID  string `xorm:"INDEX"`
+	CommentID int64
+
+	UpdatedBy int64 `xorm:"INDEX NOT NULL"`
+
+	CreatedUnix timeutil.TimeStamp `xorm:"created INDEX NOT NULL"`
```
Comment (lunny) [inline models/migrations/v1_23/v307.go]: The same as above.

**Review by lunny — APPROVED** [review]

**Comment by lunny** [comment]
Please resolve the conflict and I cannot push to your branch.

**Comment by BoYanZh** [comment]
Thank you for the notice! Done.

**Review by bohde — APPROVED** [review]

**Pull request merged**